src/prof/src/jemalloc_metrics.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+/// A thin wrapper around jemalloc::JemallocMetrics that can be used independent of jemalloc
+/// actually being used.
+
+#[cfg(all(not(target_os = "macos"), feature = "jemalloc"))]
+use crate::jemalloc;
+use mz_ore::metrics::MetricsRegistry;
+
+#[cfg(all(not(target_os = "macos"), feature = "jemalloc"))]
+pub fn register_into(registry: &MetricsRegistry) {
+    jemalloc::JemallocMetrics::register_into(registry);
+}
+
+#[cfg(not(all(not(target_os = "macos"), feature = "jemalloc")))]
+pub fn register_into(_registry: &MetricsRegistry) {
+    // No-op if we dont use jemalloc
+}
```

Comment:
This is not the correct place to have the platform conditional. That conditional should always live at the top level (either `environmentd` or `clusterd`), so that the logic for using Jemalloc or not is in one place. This crate should simply condition on whether the `jemalloc` feature is enabled.

Reply:
Ok. I'll send a fixup

Reply:
Hm, turns out I don't know how to do this. The way I did it is how the rest of `mz_prof` does (it [here](https://github.com/lluki/materialize/blob/edb876a0327dd1c820f05592035a8716fbc36f8c/src/prof/src/http/mod.rs#L154) or [here](https://github.com/lluki/materialize/blob/edb876a0327dd1c820f05592035a8716fbc36f8c/src/prof/src/lib.rs#L80)). 

I agree, it would be great to just rely on the feature and toggle this feature automatically based on the platform and use that feature as one switch. To do that we have to disable jemalloc as default feature on mac os. Afaik this is not currently possible with cargo. See https://github.com/rust-lang/cargo/issues/1197, I assume @umanwizard had the same realization. At least not without writing a custom `build.rs` which I'd prefer to avoid...

Or do I miss something obvious here?

Reply:
I think @benesch figured out how to do this in the past. It's entirely possible that I messed it up myself.

Reply:
I'll take a look this weekend.